File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        PriorityQueue<Integer> q = new PriorityQueue<>(Collections.reverseOrder());
        
        for (int i : priorities) {
            q.offer(i);
        }
        
        while (!q.isEmpty()) {
            for (int i=0; i < priorities.length; i++) {
                if (q.peek() == priorities[i]){
                    q.poll();
                    answer++;
                    
                    if (location == i) {
                        return answer;
                    }
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = []
    
    queue = deque((i, j) for i, j in enumerate(priorities))
    while queue:
        process = queue.popleft()
        
        if queue and any(process[1] < q[1] for q in queue):
            queue.append(process)
        else:
            answer.append(process)
    
    for i, j in answer:
        if i == location:
            return answer.index((i,j)) + 1